json.go

```diff
@@ -0,0 +1,357 @@
+	FD   string `json:"fd"`
+	Path string `json:"path"`
+}
+
+type SkNode struct {
+	PID         uint32       `json:"pid"`
+	OpenSockets []SocketNode `json:"open_sockets,omitempty"`
+}
+
+type SocketNode struct {
+	Protocol string `json:"protocol"`
+	Data     string `json:"data"`
+}
+
+type DisplayNode struct {
+	ContainerName      string         `json:"container_name"`
+	Image              string         `json:"image"`
+	ID                 string         `json:"id"`
+	Runtime            string         `json:"runtime"`
+	Created            string         `json:"created"`
+	Engine             string         `json:"engine"`
+	IP                 string         `json:"ip,omitempty"`
+	MAC                string         `json:"mac,omitempty"`
+	CheckpointSize     CheckpointSize `json:"checkpoint_size"`
+	CriuDumpStatistics StatsNode      `json:"criu_dump_statistics,omitempty"`
+	ProcessTree        PsNode         `json:"process_tree,omitempty"`
+	FileDescriptors    []FdNode       `json:"file_descriptors,omitempty"`
+	Sockets            []SkNode       `json:"sockets,omitempty"`
+	Mounts             []MountNode    `json:"mounts,omitempty"`
+}
+
+type MountNode struct {
+	Destination string `json:"destination"`
+	Type        string `json:"type"`
+	Source      string `json:"source"`
+}
+
+func renderJSONView(tasks []task) error {
+	var result []DisplayNode
+
+	for _, task := range tasks {
+		info, err := getCheckpointInfo(task)
+		if err != nil {
+			return err
+		}
+
+		node := DisplayNode{
+			ContainerName: info.containerInfo.Name,
+			Image:         info.configDump.RootfsImageName,
+			ID:            info.configDump.ID,
+			Runtime:       info.configDump.OCIRuntime,
+			Created:       info.containerInfo.Created,
+			Engine:        info.containerInfo.Engine,
+		}
+
+		if info.containerInfo.IP != "" {
+			node.IP = info.containerInfo.IP
+		}
+		if info.containerInfo.MAC != "" {
+			node.MAC = info.containerInfo.MAC
+		}
+
+		checkpointSizeNode := CheckpointSize{
+			CheckpointSize: metadata.ByteToString(info.archiveSizes.checkpointSize),
+		}
+
+		if info.archiveSizes.pagesSize != 0 {
+			checkpointSizeNode.MemoryPagesSize = metadata.ByteToString(info.archiveSizes.pagesSize)
+		}
+
+		if info.archiveSizes.amdgpuPagesSize != 0 {
+			checkpointSizeNode.AmdGpuMemoryPagesSize = metadata.ByteToString(info.archiveSizes.amdgpuPagesSize)
+		}
+
+		if info.archiveSizes.rootFsDiffTarSize != 0 {
+			checkpointSizeNode.RootFsDiffSize = metadata.ByteToString(info.archiveSizes.rootFsDiffTarSize)
+		}
+
+		node.CheckpointSize = checkpointSizeNode
+
+		if stats {
+			dumpStats, err := crit.GetDumpStats(task.outputDir)
+			if err != nil {
+				return fmt.Errorf("failed to get dump statistics: %w", err)
+			}
+
+			statsNode := StatsNode{
+				FreezingTime: FormatTime(dumpStats.GetFreezingTime()),
+				FrozenTime:   FormatTime(dumpStats.GetFrozenTime()),
+				MemdumpTime:  FormatTime(dumpStats.GetMemdumpTime()),
+				MemwriteTime: FormatTime(dumpStats.GetMemwriteTime()),
```

Comment:
same as above - in JSON format would be better to use the original integer value and avoid `FormatTime()`